### ISSUE TITLE
Active focus does not follow new views as they are pushed onto the stack view

### ISSUE THREAD

**Issue by yzen** [mozilla-mobile/mozilla-vpn-client#129]
We have several places where `StackView` is used:

* `main.qml`
* `StateInitialize`
* `StateMain`
* `ViewSettings`

When navigating with the keyboard and pushing a new view onto a stack, the keyboard focus remains within the previous view. Focus should really be either moved onto a new view or on the first focusable control within the view.

It looks like we can achieve this by handling view change in StackView via:

```
onCurrentItemChanged: {
    if (currentItem) {
        // Handle focus within the new view e.g. currentItem
    }
}

```

Ideally there should be a new component like `VPNStackView` that wraps the `StackView` and handles focus correctly. We would then use the new component whenever `StackView` is used.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/VPN-75)

**Comment by bakulf** [comment]
Now we have a VPNStackView object. @yzen take a look! We use it to control the 'back button' on android, but it can be extended to support what you describe here.

**Comment by yzen** [comment]
I think this is possible, we should be able to do something like:
```
```

**Comment by yzen** [comment]
Also `StateSubscriptionNeeded` still uses `StackView` directly, I'm guessing it is safe to convert to `VPNStackView` there as well?

**Comment by yzen** [comment]
Alternative solution could be setting `focus:true` on what we want to be focused by default. However, this is only a viable solution if we can figure out when view changes using a keyboard - we do not want to show keyboard focus if we clicked to open a new view.

**Comment by jess-cook03** [comment]
@lesleyjanenorton @bakulf can we confirm that this issue has been resolved? 
Also removing the UX label as this seems to be dev related primarily.
